pkg/apis/serving/metadata_validation.go

```diff
@@ -25,6 +27,17 @@ import (
 // ValidateObjectMetadata validates that `metadata` stanza of the
 // resources is correct.
 func ValidateObjectMetadata(meta metav1.Object) *apis.FieldError {
-	return apis.ValidateObjectMetadata(meta).Also(
-		autoscaling.ValidateAnnotations(meta.GetAnnotations()).ViaField("annotations"))
+	return apis.ValidateObjectMetadata(meta).
+		Also(autoscaling.ValidateAnnotations(meta.GetAnnotations()).
+			Also(validateKnativeAnnotations(meta.GetAnnotations())).
+			ViaField("annotations"))
+}
+
+func validateKnativeAnnotations(annotations map[string]string) (errs *apis.FieldError) {
+	for key := range annotations {
+		if strings.HasPrefix(key, GroupName+"/") {
```

Comment:
This feels like a const that can be extracted. No reason to compute this string every time. And it's used elsewhere.